File: algorithms/1758.minimum-changes-to-make-alternating-binary-string/solution.go
func minOperations(s string) int {
        count1 := 0
        count2 := 0

        for i, c := range s {
                if i % 2 == 0 {
                        if c == '0' {
                                count1 += 1
                        } else {
                                // c == '1'
                                count2 += 1
                        }
                } else {
                        if c == '1' {
                                count1 += 1
                        } else {
                                // c == '0'
                                count2 += 1
                        }
                }
        }

        if count1 < count2 {
                return count1
        } else {
                return count2
        }
}

File: algorithms/1758.minimum-changes-to-make-alternating-binary-string/solution.rb
# @param {String} s
# @return {Integer}
def min_operations(s)
  count1 = count2 = 0

  (0...s.length).each do |i|
    if i % 2 == 0
      if s[i] == '0'
        count1 += 1
      else
        # s[i] == '1'
        count2 += 1
      end
    else
      if s[i] == '1'
        count1 += 1
      else
        # s[i] == '0'
        count2 += 1
      end
    end
  end

  [count1, count2].min
end
